resources/packs/k8s/ingress.go

```diff
@@ -63,3 +70,63 @@ func (k *mqlK8sIngress) GetAnnotations() (interface{}, error) {
 func (k *mqlK8sIngress) GetLabels() (interface{}, error) {
 	return k8sLabels(k.MqlResource())
 }
+
+func getCertificates(ingress *networkingv1.Ingress, getSecrets func() ([]interface{}, error)) ([]interface{}, error) {
+	certificates := []interface{}{}
+	if len(ingress.Spec.TLS) > 0 {
+		// This returns ALL Secrets found in the cluster!
+		secretsInterface, err := getSecrets()
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch Secrets referenced in Ingress: %s", err)
+		}
+
+		secrets := []*mqlK8sSecret{}
+		for _, secInterface := range secretsInterface {
+			secret, ok := secInterface.(*mqlK8sSecret)
+			if !ok {
+				return nil, errors.New("returned list of Secrets failed type assertion")
+			}
+
+			secrets = append(secrets, secret)
+		}
+
+		for _, tls := range ingress.Spec.TLS {
+
+			tlsSecretFound := false
+
+			for _, secret := range secrets {
+				name, err := secret.Name()
+				if err != nil {
+					return nil, err
+				}
+
+				if tls.SecretName == name {
+					tlsSecretFound = true
+
+					certs, err := secret.GetCertificates()
+					if err != nil {
+						return nil, errors.New("error getting certificate data from Secret")
+					}
+					if certs == nil {
+						// no TLS data in Secret referenced
+						// Put a blank certificate in the list of certificates?
+						return nil, fmt.Errorf("no certificates in Secret referenced by Ingress")
+					}
+
+					certList, ok := certs.([]interface{})
+					if !ok {
+						return nil, fmt.Errorf("expected a list of Certificates")
+					}
+					certificates = append(certificates, certList...)
+				}
+			}
+			if !tlsSecretFound {
+				// Ingress references a Secret that was not found
+				// Put a blank certificate in the list of certificates?
+			}
```

Comment:
Same here. What should we do in the event the Ingress references a Secret that simply does not exist?

Reply:
I think we should do nothing. This code will run for manifests scanning. In that case I think in 99% of the time the secret won't be available in the manifest. We shouldn't crash/error